draft-ietf-dnsop-svcb-https.md

```diff
@@ -867,7 +867,32 @@ compliant recursive resolvers.  When SvcDomainName is ".", server operators
 SHOULD NOT include these hints, because they are unlikely to convey any
 performance benefit.
 
+## "mandatory" {#mandatory}
 
+In a ServiceForm RR, a SvcParamKey is considered "mandatory" if the RR will not
+function correctly for clients that ignore this SvcParamKey.  Each SVCB
+protocol mapping SHOULD specify a set of keys that are "automatically
+mandatory", i.e. mandatory if they are present in an RR.  The SvcParamKey
+"mandatory" is used to indicate any mandatory keys for this RR, in addition to
+any automatically mandatory keys that are present.
+
+A ServiceForm RR is considered "compatible" with a client if the client
+implements support for all its mandatory keys.  If the SVCB RRSet contains
+no compatible RRs, the client will generally act as if the RRSet is empty.
+
+In presentation format, "mandatory" contains a list of one or more valid
+SvcParamKeys, either by their registered name or in the unknown-key format
+({{svcfieldvalue}}).  Keys MAY appear in any order, but MUST NOT appear more
+than once.  Any listed keys MUST also appear in the SvcFieldValue.
+For example, the following is a valid SvcFieldValue:
+
+    echconfig=... key65333=ex1 key65444=ex2 mandatory=key65444,echconfig
+
+In wire format, the keys are represented by their numeric values in
+network byte order, concatenated in ascending order.
+
+This SvcParamKey is always automatically mandatory, and MUST NOT appear in its
+own value list.
```

Comment:
Perhaps reframe as:

"This SvcParamKey is always automatically mandatory.  Automatically mandatory entries MUST NOT appear in the value list."

(Unless we want to allow them to appear as doing so is idempotent?)

Reply:
I don't want to make this "MUST NOT", because zone file parsers can't detect it (they are not scheme-aware), and I would like to avoid introducing syntax errors that are caught by clients but not by the zone file parser.